app/scripts/setup-plugins.js

```diff
@@ -37,14 +38,32 @@ async function extractArchive(
         }
 
         // Move the plugins contents to the plugins folder
-        fs.copyFileSync(
-          path.join(tmpFolder, 'package', 'dist', 'main.js'),
-          path.join(pluginFolder, 'main.js')
-        );
-        fs.copyFileSync(
-          path.join(tmpFolder, 'package', 'package.json'),
-          path.join(pluginFolder, 'package.json')
-        );
+        const mainLocationExpr = path.join(tmpFolder, '*', 'main.js');
+        const mainLocations = glob.sync(mainLocationExpr);
```

Comment:
why do we need this pattern match?

Reply:
Because AFAIU the tarballs generated will unpack into PLUGIN_NAME/main.js and PLUGIN_NAME/package.js .
We don't know what PLUGIN_NAME is at this point.

Reply:
Cool got it, makes sense.